astronomer/providers/amazon/aws/operators/sagemaker.py

```diff
@@ -151,8 +151,17 @@ def execute(self, context: Context) -> None:  # type: ignore[override]
         self.preprocess_config()
         model_config = self.config.get("Model")
         transform_config = self.config.get("Transform", self.config)
-        if self.check_if_job_exists:
-            self._check_if_transform_job_exists()
+        if self.check_if_job_exists:  # pragma: no cover
+            try:
+                # for apache-airflow-providers-amazon<=7.2.1
+                self._check_if_transform_job_exists()
+            except AttributeError:
+                # for apache-airflow-providers-amazon>=7.3.0
+                transform_config["TransformJobName"] = self._get_unique_job_name(  # type: ignore[attr-defined]
+                    transform_config["TransformJobName"],
+                    self.action_if_job_exists == "fail",
+                    self.hook.describe_transform_job,
+                )
```

Comment:
In my opinion, the code for latest provider version should be tried first and if it fails we should handle old versions in the except block. WDYT?

Reply:
till now we have used same convention https://github.com/astronomer/astronomer-providers/blob/main/astronomer/providers/amazon/aws/operators/emr.py#L46-L51, https://github.com/astronomer/astronomer-providers/blob/main/astronomer/providers/databricks/operators/databricks.py#L346-L351
so i maintained same way here also.

Reply:
have created a ticket to make these changes in other places also. https://github.com/astronomer/astronomer-providers/issues/913

Reply:
yeah added the ticket https://github.com/astronomer/astronomer-providers/issues/913